src/modules/User/resolvers.ts

```diff
@@ -0,0 +1,70 @@
+import { queryField, mutationField, nonNull } from 'nexus/dist';
+import { compare, hash } from 'bcrypt';
+import { sign } from 'jsonwebtoken';
+import { AUTH_SIGNUP_ENABLED, JWT_EXPIRATION_PERIOD, JWT_SECRET } from '../../env';
+
+export const me = queryField('me', {
+  type: 'User',
+  resolve: (_, __, ctx) => {
+    if (!ctx.authVerificationResult || !('userId' in ctx.authVerificationResult)) {
+      throw new Error('Not authorized');
+    }
+    return ctx.prisma.user.findUnique({
+      where: {
+        id: ctx.authVerificationResult.userId,
+      },
+    });
+  },
+});
+
+export const signIn = mutationField('signIn', {
+  type: 'AuthPayload',
+  args: {
+    user: nonNull('UserNamePass'),
+  },
+  resolve: async (_parent, { user: userNamePass }, ctx) => {
+    const { username, password } = userNamePass;
+    const user = await ctx.prisma.user.findUnique({
+      where: {
+        username,
+      },
+    });
+    if (!user) {
+      throw new Error('user not found');
+    }
+    const passwordValid = (await compare(password, user.password || '')) || false;
+    if (!passwordValid) {
+      throw new Error('invalid password');
+    }
+    return {
+      token: sign({ userId: user.id }, JWT_SECRET, { expiresIn: JWT_EXPIRATION_PERIOD }),
+      user,
+    };
```

Comment:
One big advice we gave while looking into the [`ecosystem-api`](https://github.com/makerdao-ses/ecosystem-api) is to not spread the logic between model and the resolvers. Ideally, _all business logic_ should stay inside the model and the resolves should only define available queries/mutations and call a single method of the model. 

See second point of the [typeorm vs prisma discussion](https://github.com/makerdao-ses/switchboard-boilerplate/issues/2#issuecomment-1456504670) for the approaches on how to make this logic part of the model. To make it clear: I propose to replace all this code with `return User.signIn(userNamePass)` or `return ctx.prisma.user.signIn(userNamePass)`

(here and in all other resolvers)